modules/agentless-scanner-role/main.tf

```diff
@@ -47,22 +47,22 @@ data "aws_iam_policy_document" "scanner_policy_document" {
   }
 
   dynamic "statement" {
-    for_each = var.api_key_secret_arn != null ? [1] : []
+    for_each = var.api_key_secret_arns
```

Comment:
Nit: We don't need to do `dynamic` and `for_each` here since we have a list of ARN we can put directly in `resources`